src/payment/strategies/googlepay/googlepay-payment-processor.ts

```diff
@@ -134,6 +113,26 @@ export default class GooglePayPaymentProcessor {
                 ]).then(([googlePay, paymentDataRequest]) => {
                     this._googlePayClient = this._getGooglePayClient(googlePay, testMode);
                     this._paymentDataRequest = paymentDataRequest;
+
+                    return this._googlePayClient.isReadyToPay({
+                        allowedPaymentMethods: [
+                            {
+                                type: paymentDataRequest.allowedPaymentMethods[0].type,
+                                parameters: {
+                                    allowedAuthMethods: paymentDataRequest.allowedPaymentMethods[0].parameters.allowedAuthMethods,
+                                    allowedCardNetworks: paymentDataRequest.allowedPaymentMethods[0].parameters.allowedCardNetworks,
+                                },
+                            },
+                        ],
+                        apiVersion: paymentDataRequest.apiVersion,
+                        apiVersionMinor: paymentDataRequest.apiVersionMinor,
+                    }).then(response => {
+                        if (response.result) {
+                            Promise.resolve();
```

Comment:
You can just do a `return` if you don't want the promise to have a resolved value. i.e.:

```ts
if (response.result) {
    return;
}
```